src/components/Card.js

```diff
@@ -1,34 +1,79 @@
 class Card {
-  constructor(cardData, cardSelector, handleCardImageClick) {
+  constructor(
+    cardData,
+    cardSelector,
+    handleCardImageClick,
+    handleDeleteCard,
+    handleLikeCard,
+    handleDeleteLike,
+    confirmDeleteCardPopup,
+    userId
+  ) {
     this._name = cardData.name;
     this._link = cardData.link;
+    this._likes = cardData.likes || [];
     this._handleCardImageClick = handleCardImageClick;
     this._cardSelector = cardSelector;
+    this._cardId = cardData._id;
+    this._handleDeleteCard = handleDeleteCard;
+    this._handleLikeCard = handleLikeCard;
+    this._handleDeleteLike = handleDeleteLike;
+    this._confirmDeletePopup = confirmDeleteCardPopup;
+    this._userId = userId;
+    this._isLiked = this._likes.some((like) => like._id === this._userId);
```

Comment:
`this._isLiked`


You should check `cardData.isLiked` inside the `Card` class, to see if the card is liked

```js
    this._isLiked = cardData.isLiked;


```